### PR TITLE
[JENKINS-53913] Make graceful shutdown timeout customizable.

### PR DESCRIPTION
**Pull request by rudolfwalter**
Let the user change the currently hard-coded timeout value from 3 minutes to whatever they want.

https://issues.jenkins-ci.org/browse/JENKINS-53913

### CODE REVIEW

**Comment by pjdarton** [comment]
Sure; I'm broadly in favor of this sort of thing.
I just can't promise a fast review/merge due to current work commitments right now.

**Comment by rudolfwalter** [comment]
If you'll forgive me asking, do you have an idea when you'll be able to take a look at this? By the end of the year? Next year?

File: src/main/java/org/jenkinsci/plugins/vsphere/tools/VSphere.java [hunk 1]
```diff
@@ -904,7 +904,12 @@ public boolean vmToolIsEnabled(VirtualMachine vm) {
         return ((status == VirtualMachineToolsStatus.toolsOk) || (status == VirtualMachineToolsStatus.toolsOld));
     }
 
+    @Deprecated
```
[the remark this answers is not in the document]
Comment (rudolfwalter) [reply src/main/java/org/jenkinsci/plugins/vsphere/tools/VSphere.java]: Done.

File: src/main/java/org/jenkinsci/plugins/vsphere/tools/VSphere.java [hunk 2]
```diff
@@ -904,7 +904,12 @@ public boolean vmToolIsEnabled(VirtualMachine vm) {
...
     public void powerOffVm(VirtualMachine vm, boolean evenIfSuspended, boolean shutdownGracefully) throws VSphereException {
+        powerOffVm(vm, evenIfSuspended, shutdownGracefully ? 180 : 0);
+    }
+
+    public void powerOffVm(VirtualMachine vm, boolean evenIfSuspended, int gracefulShutdownSeconds) throws VSphereException {
```
[the remark this answers is not in the document]
Comment (rudolfwalter) [reply src/main/java/org/jenkinsci/plugins/vsphere/tools/VSphere.java]: Done.

**Comment by rudolfwalter** [comment]
Below are two screenshots showing the relevant UI:
capture1
capture2

**Comment by rudolfwalter** [comment]
@pjdarton Could we get a new release that includes this? Thanks :)

**Comment by pjdarton** [comment]
Yes, I can do that ... if you can help test it prior to release - there's so little unit-testing on this plugin that I prefer to do a bit of manual testing before a release, and as different folks use the plugin differently, it's very useful to have more than one person do a bit of testing.

First, I've got some cosmetic changes that I'd like to include in the next release which I need to put into a PR.  Once they're in a PR, we can both test the result and, if it works to our mutual satisfaction, I'll merge those changes and then I'll push out a new release.

**Comment by pjdarton** [comment]
OK, I've created a PR #101
Grab the build from https://ci.jenkins.io/job/Plugins/job/vsphere-cloud-plugin/job/PR-101/lastSuccessfulBuild/artifact/target/ and try it out where you work.
I'll also test it locally.
If everything checks out, I'll merge and release.

**Comment by pjdarton** [comment]
My own testing convinced me that my cosmetic changes (#101) were safe to merge, so I've merged them.
@rudolfwalter  Can you please test https://ci.jenkins.io/job/Plugins/job/vsphere-cloud-plugin/job/master/lastSuccessfulBuild/artifact/target/ where you work and let me know the outcome?

The sooner you get back to me with test results, the sooner I'll push out a release...

**Pull request merged**